moonlight-qt: 6.0.1 -> 6.1.0

azuwis opened the pull request:
- Use finalAttrs pattern
- Add meta.changelog
 
Changelog: https://github.com/moonlight-stream/moonlight-qt/releases/tag/v6.1.0

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

momeemt commented:
Result of `nixpkgs-review pr 343717` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>moonlight-qt</li>
  </ul>
</details>

momeemt reviewed (changes requested):
Thanks for your contribution. I left the comment about using `with lib;`.

momeemt commented:
Result of `nixpkgs-review pr 343717` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>moonlight-qt</li>
  </ul>
</details>

momeemt commented:
> error: builder for '/nix/store/9cbkvnwpg3zw8pcvr4wlhd8wapm0dh5w-source.drv' failed wi
th exit code 1;
       last 7 log lines:
       > exporting https://github.com/moonlight-stream/moonlight-qt.git (rev v6.1.0)
into /nix/store/8njhghs1bkpcr57pkf0wisbp3pv3w86z-source
       > Initialized empty Git repository in /nix/store/8njhghs1bkpcr57pkf0wisbp3pv3w
86z-source/.git/
       > fatal: unable to access 'https://github.com/moonlight-stream/moonlight-qt.gi
t/': OpenSSL/3.0.14: error:16000069:STORE routines::unregistered scheme
       > fatal: unable to access 'https://github.com/moonlight-stream/moonlight-qt.gi
t/': OpenSSL/3.0.14: error:16000069:STORE routines::unregistered scheme
       > fatal: unable to access 'https://github.com/moonlight-stream/moonlight-qt.gi
t/': OpenSSL/3.0.14: error:16000069:STORE routines::unregistered scheme
       > fatal: unable to access 'https://github.com/moonlight-stream/moonlight-qt.gi
t/': OpenSSL/3.0.14: error:16000069:STORE routines::unregistered scheme
       > Unable to checkout refs/tags/v6.1.0 from https://github.com/moonlight-stream
/moonlight-qt.git.
       For full logs, run 'nix-store -l /nix/store/9cbkvnwpg3zw8pcvr4wlhd8wapm0dh5w-s
ource.drv'.
error: 1 dependencies of derivation '/nix/store/z8168v1g4zn94jbwspffw3p138gd5fab-moon
light-qt-6.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hmd20val5rw3cpc5f8ds5nzikqdbl590-revi
ew-shell.drv' failed to build

I got the above errors, but it may be caused by my local environment. I hope you don't mind the previous result of nixpkgs-review. (I will investigate the reason.)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4680

azuwis committed:
moonlight-qt: 6.0.1 -> 6.1.0

- Bump apple sdk to 12.3 to fix build error `error: expected ';' after top level declarator`
- Use finalAttrs pattern
- Add meta.changelog
- Remove `with lib` usage

Changelog: https://github.com/moonlight-stream/moonlight-qt/releases/tag/v6.1.0

SuperSandro2000 commented on pkgs/by-name/mo/moonlight-qt/package.nix:
```diff
@@ -35,23 +35,23 @@ let
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-zrl8WPXvQ/7FTqFnpwoXEJ85prtgJWoWNsdckw5+JHI=";
+    repo = finalAttrs.pname;
```
```suggestion
    repo = "moonlight-qt";
```

momeemt commented:
Result of `nixpkgs-review pr 343717` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>moonlight-qt</li>
  </ul>
</details>

momeemt commented:
I have reinstalled Nix and its related softwares, but it does not build in my local.
Nix throws the following errors, different from the previous ones.

```
1 package updated:
moonlight-qt (6.0.1 → 6.1.0)

$ nix build --file /nix/store/vixjgcnc8br92cnf4n3vyzq7fklakfih-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-she
ll.nix --nix-path 'nixpkgs=/Users/momeemt/.cache/nixpkgs-review/pr-343717-2/nixpkgs nixpkgs-overlays=/var/folders/09/9ttjqns51hx4cl_3c96llt_r00
00gp/T/tmp40c_0jlh' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --args
tr system aarch64-darwin --argstr nixpkgs-path /Users/momeemt/.cache/nixpkgs-review/pr-343717-2/nixpkgs --argstr nixpkgs-config-path /var/folde
rs/09/9ttjqns51hx4cl_3c96llt_r0000gp/T/tmpzgk9i3ll.nix --argstr attrs-path /Users/momeemt/.cache/nixpkgs-review/pr-343717-2/attrs.nix
error: builder for '/nix/store/0pa9p1a79vma071ffg3qfd3sq3lc1h4m-moonlight-qt-6.1.0.drv' failed with exit code 2;
       last 10 log lines:
       > streaming/streamutils.cpp:203:22: warning: comparison of integers of different signs: 'int' and 'uint32_t' (aka 'unsigned int') [-Wsig
n-compare]
       >     if (displayIndex >= displayCount) {
       >         ~~~~~~~~~~~~ ^  ~~~~~~~~~~~~
       > 1 warning and 2 errors generated.
       > make[2]: *** [Makefile.Release:4268: release/streamutils.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[2]: Leaving directory '/private/tmp/nix-build-moonlight-qt-6.1.0.drv-0/source/app'
       > make[1]: *** [Makefile:45: release] Error 2
       > make[1]: Leaving directory '/private/tmp/nix-build-moonlight-qt-6.1.0.drv-0/source/app'
       > make: *** [Makefile:107: sub-app-make_first] Error 2
       For full logs, run 'nix-store -l /nix/store/0pa9p1a79vma071ffg3qfd3sq3lc1h4m-moonlight-qt-6.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/gak1c5g8n9wq2pgd3aid1clklzmjasj5-review-shell.drv' failed to build
```
